src/drivers/wrappers/tests/tests.rs

```diff
@@ -31,6 +31,15 @@ fn test_slice_reader_returns_zero_for_empty_buf() {
     assert_eq!(0, driver.pread(&mut buf, 1).unwrap());
 }
 
+#[test]
+fn test_slice_reader_returns_number_of_bytes_read() {
+    let data = [1, 2, 3, 4, 5];
+    let driver = SliceReader::new(&data);
+    let mut buf = [0; 10];
+
+    assert_eq!(5, driver.pread(&mut buf, 0).unwrap());
+}
```

Comment:
Worth also testing contents?

    assert_eq!(&buf[..5], &data);
    assert_eq!(&buf[5..], &[0; 5]);

Reply:
That is a good idea, thanks!